core/os/os_linux.odin

```diff
@@ -253,6 +256,29 @@ X_OK :: 1; // Test for execute permission
 W_OK :: 2; // Test for write permission
 R_OK :: 4; // Test for read permission
 
+passwd :: struct {
+	pw_name: cstring,
+	pw_passwd: cstring,
+	pw_uid: uid_t,
+	pw_gid: gid_t,
+	pw_gecos: cstring,
+	pw_dir: cstring,
+	pw_shell: cstring,
+}
+
+DIR :: opaque [0]byte;
```

Comment:
Probably better to either declare this here, or to use the _actual_ size for the array, so it's at least the right size.

Or is the struct _supposed_ to never be used directly?
You could make it `DIR :: distinct rawptr;` in that case, which would be the same style as `os.Handle`.

Reply:
I think that DIR is supposed to be an opaque type. I'm not sure though. I couldn't find any definition for __dirstream (which is what DIR is defined as).
I actually originally used `DIRP :: distinct rawptr`, but them someone on discord, either you or Oskar, mentioned about the `opaque` thing, so I thought I'd use that to be more consistent.